### DISCUSSION TITLE
save the webpage

### DISCUSSION THREAD

**Discussion by slvamegestic** [seleniumbase/SeleniumBase#2391] [General]
How can I save the webpage as a single html file?
```
sb.get_page_source()
```
above code only saves the html string instead of css styles. i need to extract the exact html with styles. Do any idea how to achieve this selenium base?
I am using selenium base with python3

**Comment by mdmintz** [comment]
Basic example of saving HTML into a file:

```python
from seleniumbase import SB

with SB(browser="chrome") as sb:
    sb.open("data:text/html,<div>Hello<br><input></div>")
    source = sb.get_page_source()
    sb.save_data_as(source, "page_source.html", "./")
```

`sb.driver.page_source` is the same as `sb.get_page_source()`:

```python
from seleniumbase import SB

with SB(browser="chrome") as sb:
    sb.open("data:text/html,<div>Hello<br><input></div>")
    source = sb.driver.page_source
    sb.save_data_as(source, "page_source.html", "./")
```

For any other ways of getting the page source, try [Stack Overflow](https://stackoverflow.com/).
Saving the page source with styles included isn't specific to SeleniumBase.

**Reply by slvamegestic** [reply]
Ok thanks, I will take a look for Page source with styles included
